### PR TITLE
CI: use ruff to format xdsl

### PR DESCRIPTION
**Pull request by superlopuh**
Runs in `0.04s` on my computer.

The Ruff format is designed to be a drop-in replacement for black. As you see, most of the diff is adding a new line at the start of every class, which seems like a good change. I'll comment on the config files with changes for easier reviewing.

Note stacked PR.

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (superlopuh) [inline .pre-commit-config.yaml:1]: Config change 1/3

File: pyproject.toml
Comment (superlopuh) [inline pyproject.toml:1]: Config change 2/3

File: xdsl/utils/comparisons.py [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
+
+The main idea is to not have the signedness be a property of the type of the value,
+and rather be a property of the operation. That means that a signless value can be
+interpreted as either a signed or unsigned value at runtime, depending on the
+operation that acts on it.
+
+During interpretation, this gets a little tricky, as the same bit pattern can be
+interpreted as two runtime values, meaning that comparing signless values is a little
+involved. For example, a signless value of 5 is equal to a signless value of -3, since
+their bit representations are the same.
```
Comment (PapyChacal) [inline xdsl/utils/comparisons.py:32]: :heart:

**Review by PapyChacal — APPROVED** [review]
Nice, just a comment on the underlying one about lines length:
- #3022

**Pull request merged**